libimobiledevice: update all packages to the latest git version

httpstorm opened the pull request:
Due to dependency, all packages need to be updated together. The latest releases are available on GitHub. Some of the packages have not received a release since 2020-06-16. They had to be updated to the last commit. Added 3 new packages for completeness.

For `usbmuxd`, revert a change from the following commit, otherwise iPhone tethering will not work.
https://github.com/libimobiledevice/usbmuxd/commit/c7a0dd9b82633ea347497626282e3051a469ef50#r141306029

Maintainer: @neheb 
Compile tested:
- Marvell EBU Armada, Marvell Armada 37x/38x/XP, Linksys WRT3200ACM, [main](https://github.com/openwrt/openwrt/commit/e0363233c958cf591597006f6a0853dc1b0a012f)
- Atheros ATH79, Generic, TP-Link TL-WR1043N/ND v4, main
- x86, x86_64, Generic x86/64, main

Run tested:
- Marvell EBU Armada, Marvell Armada 37x/38x/XP, Linksys WRT3200ACM

Please let me know if any changes are needed.
@nbd168 @Ansuel @robimarko @hauke @1715173329 @jow-

1715173329 commented on libs/libimobiledevice-glue/Makefile:
```diff
@@ -0,0 +1,69 @@
+PKG_RELEASE:=1
+
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.bz2
+PKG_SOURCE_URL:=https://github.com/libimobiledevice/$(PKG_NAME)/releases/download/$(PKG_VERSION)
```
```suggestion
PKG_SOURCE_URL:=https://github.com/libimobiledevice/libimobiledevice-glue/releases/download/$(PKG_VERSION)
```

httpstorm replied:
Thanks! I applied this to all packages.

1715173329 commented on libs/libideviceactivation/Makefile:
```diff
@@ -0,0 +1,99 @@
+PKG_VERSION:=1.1.2
+PKG_RELEASE:=1
+
+PKG_SOURCE_URL:=https://github.com/libimobiledevice/$(PKG_NAME).git
+PKG_SOURCE_PROTO:=git
```
```suggestion
PKG_SOURCE_PROTO:=git
PKG_SOURCE_URL:=https://github.com/libimobiledevice/libideviceactivation.git
```
please leave repo url as-is for better readability
the same applies below

httpstorm replied:
I applied this to all packages: PROTO before URL, URL as is.

1715173329 commented on libs/libideviceactivation/Makefile:
```diff
@@ -0,0 +1,99 @@
+#
+# Copyright (C) 2012-2024 OpenWrt.org
```
please either assign the copyright to yourself or simply remove it
the same applies below

httpstorm replied:
I removed it. I giveaway any copyright in hope to improve OpenWRT and help others. 😀
I also adjusted license file and version to make sure they match each package. I just noticed they vary.

neheb commented on utils/ifuse/Makefile:
```diff
@@ -0,0 +1,56 @@
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=ifuse
+PKG_VERSION:=1.1.5
```
Does PKG_VERSION make sense here?

httpstorm replied:
For those packages that have no release and were switched to the latest commit, `PKG_VERSION` defines the string returned by `--version`, for example:

``` bash
ifuse --version

# PKG_VERSION:=1.1.5
ifuse 1.1.5

# PKG_VERSION not set in Makefile, seems to be auto-generated
ifuse 2023.01.06~814a0e38
```

The [commit](https://github.com/libimobiledevice/ifuse/commit/38c9f456fb4fc007ea99dff70eda32950a39fd2a) right after the last release says *Post release version bump to 1.1.5*. So if we honour to display that version, we need to set it in `PKG_VERSION`. If we prefer the auto-generated version, we should not set `PKG_VERSION`.

Note that `PKG_VERSION` is later used for `PACKAGE_VERSION` which is required for the build when we use `autoreconf`.
``` Makefile
CONFIGURE_ARGS += \
	PACKAGE_VERSION=$(PKG_VERSION)
```

Let me know if I should keep it or remove it.

neheb commented on utils/usbmuxd/Makefile:
```diff
@@ -8,18 +8,21 @@
 PKG_NAME:=usbmuxd
-PKG_VERSION:=1.1.1
-PKG_RELEASE:=2
+PKG_VERSION:=1.1.2
```
I don’t think PKG_VERSION makes sense here.

httpstorm replied:
@neheb 
Thanks Rosen! Indeed having date and commit returned by `--version` makes more sense. I updated all 6 packages that are now using a git commit. I rebuilt and tested them. Here's how the version is reported now:
``` bash
plistutil 2.4.0
inetcat 2.1.0
ideviceinfo 2024.04.16~5f083426
idevice_id 2024.04.16~5f083426
usbmuxd # no version reported no mater if PKG_VERSION is set or not
irecovery 1.2.0
idevicerestore 2024.04.04~6d40d0ab
ifuse 2023.01.06~814a0e38
ideviceinstaller 2024.01.15~22872c35
ideviceactivation 2024.04.16~6925d58e
```

httpstorm commented:
@neheb 
Any idea why all these tests suddenly failed? I recompiled all 10 packages `{clean,compile}`, then `make -j 16` and re-flashed my router. I'm still online, so all is good.

httpstorm commented:
@neheb @1715173329 
I fixed `PKG_MIRROR_HASH`, also fixed `usbmuxd` to print the version.
Locally everything builds fine and I ran `make check` for all affected packages.

Can you help me? Why am I getting this error?
```
This step has been truncated due to its large size. Download the full logs from the  menu once the workflow run has completed.
```

httpstorm commented:
@neheb @1715173329 
I rebased the branch on master just so I can trigger the tests again and now all tests pass. I wonder why?
Is there anything else I need to change?

1715173329 commented:
`gettext-full` exploded in previous CI build (guessing caused by outdated SDK image?), and now it works again.
It's irrelevant to this PR.
